ludwig/modules/metric_registry.py

```diff
@@ -0,0 +1,29 @@
+from typing import List, Union
+
+from ludwig.utils.registry import Registry
+
+metric_feature_registry = Registry()
+metric_registry = Registry()
+
+
+def register_metric(name: str, features: Union[str, List[str]]):
```

Comment:
This came from my attempt to address the circular dependency introduced by the `"OutputFeature"` type hint on `LudwigMetric.can_report()`, added in https://github.com/ludwig-ai/ludwig/pull/1522/files. The linter complains (rightly so) that it doesn't know what OutputFeature is.

The circular dependency arises because `base_feature.py` needs the metrics registry, and metrics registry was in `metrics_modules.py`, but `metrics_modules.py` needs to depend on `OutputFeature` in `base_feature.py` for the type hint.

My experience is that circular dependencies are usually an indication of an opportunity for a better design, so that's my hope with splitting out the metrics registry into `metric_registry.py` file, separate from actual metric definitions.